### PR TITLE
monitoring/grafana: improve grafana unit tests variable substitution

### PR DESCRIPTION
**Pull request by pereman2**
Promql tests were failing locally because of wrong matching with regex, I haven't seen this problem on master but just in case let's improve this matching expression.
Signed-off-by: Pere Diaz Bou <pdiazbou@redhat.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by avanthakkar — APPROVED** [review]
LGTM!

File: monitoring/grafana/dashboards/tests/util.py [hunk 1]
```diff
@@ -66,3 +66,26 @@ def add_dashboard_variables(data: Dict[str, Any], dashboard_data: Dict[str, Any]
...
+    >>> replace_grafana_expr_variables('metric{name~="$osd_hosts|$other|$osd"}', \
+        'other', 'replacement')
+    'metric{name~="$osd_hosts|replacement|$osd"}'
+
+    It replaces words with dollar prefix
+    >>> replace_grafana_expr_variables('metric{name~="no_dollar|$other|$osd"}', \
+        'no_dollar', 'replacement')
+    'metric{name~="no_dollar|$other|$osd"}'
+    """
+    import re
```
Comment (alfonsomthd) [inline monitoring/grafana/dashboards/tests/util.py:88]: From [PEP8](https://www.python.org/dev/peps/pep-0008/#imports):
```Imports are always put at the top of the file, just after any module comments and docstrings, and before module globals and constants.```

**Review by alfonsomthd — APPROVED** [review]

File: monitoring/grafana/dashboards/tests/util.py (hunk 1 above)
Comment (pereman2) [reply monitoring/grafana/dashboards/tests/util.py]: done

**Review by nizamial09 — APPROVED** [review]

File: monitoring/grafana/dashboards/tox.ini [hunk 2]
```diff
@@ -41,4 +41,5 @@ deps =
 depends = grafonnet-check
 setenv =
 commands =
+    python -m doctest tests/util.py
```
Comment (epuertat) [inline monitoring/grafana/dashboards/tox.ini:44]: Nit: if we are running pytest (are we?), this is as simple as:
```ini
[pytest]
addopts = --doctest-modules
```

**Comment by epuertat** [comment]
@pereman2 if we want to backport this to Pacific we need a tracker. Is this the issue that happened in ARM only? If so, you can clone that tracker and use it for backporting.

**Pull request merged**